### PR TITLE
fixed some warnings

### PR DESCRIPTION
**Pull request by cdietrich**
fixed some warnings
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

**Comment by pisv** [comment]
Thanks for the clean-up, Christian!

File: org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java [hunk 1]
```diff
@@ -295,7 +295,7 @@ protected Object parseBody(JsonReader in, String messageType, int request_seq, S
 		} else if ("response".equals(messageType) && success != null && success) {
 			return super.parseResult(in, Integer.toString(request_seq));
 		} else {
-			return new JsonParser().parse(in);
+			return JsonParser.parseReader(in);
```
Comment (jonahgraham) [inline org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: This change needs a corresponding change in the manifest ranges for Gson. Gson does not (strictly) follow Semantic versioning and this new method (and the deprecation of what was removed) was introduced in 2.8.6. Our current code reads:

```
Import-Package: com.google.gson;version="[2.8,3)"
```

However, I don't know what magic generates that range.

Note that the Eclipse platform had this issue and it was fixed just before 2021-03 release https://bugs.eclipse.org/bugs/show_bug.cgi?id=571550

This applies to many of the changes below, I am not marking each.

**Review by jonahgraham — CHANGES_REQUESTED** [review]

File: org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: good point. the only reference i found was the 2.8.6

Comment (cdietrich) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: wonder if this comes from the old gradle osgi plugin

Comment (jonahgraham) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: Adding this above existing instuction is close - but not quite right in https://github.com/eclipse/lsp4j/blob/ad99a3d77139351dbc00c2dacd388fe749652309/org.eclipse.lsp4j.jsonrpc.debug/build.gradle#L22-L24
```
	instruction 'Import-Package', "com.google.gson.*;version=\"$versions.gson\""
```

For some reason the above causes downstream bundles in gradle to fail to build for me, and the above is a lower bound, but there should be an upper bound. The above generates:

```
Import-Package: com.google.gson;version="2.8.6",
```

Comment (cdietrich) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: something like for guava? do we assume nothing breaks with 2.9 etc?

Comment (jonahgraham) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: Yes I think you are right, this needs updating too:

https://github.com/eclipse/lsp4j/blob/da194124f75c8b18be15a3c9cab03f6b62fdb92a/gradle/versions.gradle#L19

like 

https://github.com/eclipse/lsp4j/blob/da194124f75c8b18be15a3c9cab03f6b62fdb92a/gradle/versions.gradle#L17

Comment (cdietrich) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: the question is: should the upper bound be 2.9 or 3?

File: releng/lsp4j-feature/feature.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@ SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
    </license>
 
    <requires>
-      <import plugin="com.google.gson" version="2.8.0" match="equivalent"/>
+      <import plugin="com.google.gson" version="2.8.6" match="equivalent"/>
```
Comment (jonahgraham) [inline releng/lsp4j-feature/feature.xml:29]: thanks for catching this part

File: org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java (hunk 1 above)
Comment (jonahgraham) [reply org.eclipse.lsp4j.jsonrpc.debug/src/main/java/org/eclipse/lsp4j/jsonrpc/debug/adapters/DebugMessageTypeAdapter.java:298]: We are saying the same thing at the same time :-) I think this is resolved now with your version fixes.

**Review by jonahgraham — APPROVED** [review]

**Comment by cdietrich** [comment]
is org.eclipse.lsp4j.jsonrpc.test.IntegrationTest.testMalformedJson2 flaky?

**Comment by cdietrich** [comment]
@eclipse-lsp4j-bot run tests

**Comment by jonahgraham** [comment]
> is org.eclipse.lsp4j.jsonrpc.test.IntegrationTest.testMalformedJson2 flaky?

I don't think so - passing consistently on master https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/master/

**Comment by jonahgraham** [comment]
> > is org.eclipse.lsp4j.jsonrpc.test.IntegrationTest.testMalformedJson2 flaky?
> 
> I don't think so - passing consistently on master https://ci.eclipse.org/lsp4j/job/lsp4j-multi-build/job/master/

Well - it may pass a lot, but the test does have a race condition. See #531

**Pull request merged**